core/src/test/java/org/elasticsearch/search/internal/InternalSearchHitTests.java

```diff
@@ -94,4 +215,45 @@ public void testHasSource() {
         searchHit.sourceRef(new BytesArray("{}"));
         assertTrue(searchHit.hasSource());
     }
+
+    public void testParseShardId() {
```

Comment:
I followed the suggestion of @tlrx to do the shrd id parsing in "fromString" in ShardId along side of the "toString()" method. I think this is good because because the proximity clearly communicates that one belongs to the other, so I'd like to keep it there. As for the location of the test: Ideally this would be placed in a new `ShardIdTests` but that class should then have some more tests. I'd prefer to do this in a follow up PR.